core/src/local/mod.rs

```diff
@@ -16,10 +55,17 @@ impl Tag {
     }
 }
 
+#[derive(Clone)]
 pub struct Row(pub(crate) Rc<Vec<DataType>>);
 
 unsafe impl Send for Row {}
 
+impl Row {
+    pub fn unpack(self) -> Vec<DataType> {
+        Rc::try_unwrap(self.0).unwrap()
+    }
+}
+
```

Comment:
I'm a little sad about the indirection through `Rc` for `PersistentState`, but I guess it's okay..